quickwit/quickwit-integration-tests/src/test_utils/cluster_sandbox.rs

```diff
@@ -187,6 +188,68 @@ impl ClusterSandbox {
         Ok(())
     }
 
+    // Waits for the needed number of indexing pipeline to start.
+    pub async fn wait_for_indexing_pipelines(
+        &self,
+        required_pipeline_num: usize,
+    ) -> anyhow::Result<()> {
+        let mut num_attempts = 0;
+        let max_num_attempts = 3;
+        while num_attempts < max_num_attempts {
+            if num_attempts > 0 {
+                tokio::time::sleep(Duration::from_millis(100 * (num_attempts))).await;
+            }
+            if self
+                .indexer_rest_client
+                .node_stats()
+                .indexing()
+                .await
+                .unwrap()
+                .num_running_pipelines
+                == required_pipeline_num
+            {
+                return Ok(());
+            }
+            num_attempts += 1;
+        }
+        if num_attempts == max_num_attempts {
+            anyhow::bail!("Too many attempts to get expected number of pipelines.");
+        }
+        Ok(())
```

Comment:
const MAX_NUM_ATTEMPTS: u64 = 3;
        for num_attempts in 0..MAX_NUM_ATTEMPTS {
            if num_attempts > 0 {
                tokio::time::sleep(Duration::from_millis(100 * num_attempts)).await;
            }
            if self
                .indexer_rest_client
                .node_stats()
                .indexing()
                .await
                .unwrap()
                .num_running_pipelines
                == required_pipeline_num
            {
                return Ok(());
            }
        }
        anyhow::bail!("Too many attempts to get expected number of pipelines.")